File: verdict-backend/src/routes/briefs.ts
import type { FastifyInstance, FastifyRequest, FastifyReply } from 'fastify';
import { db } from '../lib/db.js';
import { requireAuth } from '../middleware/auth.js';

export async function briefsRoutes(app: FastifyInstance) {
  app.get<{ Params: { briefId: string } }>(
    '/:briefId',
    { preHandler: [requireAuth] },
    async (request, reply) => {
      const user = request.user!;
      const { briefId } = request.params;
      const brief = await db.brief.findFirst({
        where: { id: briefId, firmId: user.firmId },
      });
      if (!brief)
        return reply.code(404).send({ success: false, error: { code: 'NOT_FOUND', statusCode: 404 } });

      return reply.send({
        success: true,
        data: {
          id: brief.id,
          sessionScore: brief.sessionScore,
          consistencyRate: brief.consistencyRate,
          deltaVsBaseline: brief.deltaVsBaseline,
          confirmedFlags: brief.confirmedFlags,
          objectionCount: brief.objectionCount,
          composureAlerts: brief.composureAlerts,
          topRecommendations: brief.topRecommendations,
          narrativeText: brief.narrativeText,
          weaknessMapScores: brief.weaknessMapScores,
          shareToken: brief.shareToken,
          pdfS3Key: brief.pdfS3Key,
          createdAt: brief.createdAt,
        },
      });
    }
  );

  // Public share endpoint — no auth required
  app.get<{ Params: { shareToken: string } }>(
    '/share/:shareToken',
    async (request, reply) => {
      const { shareToken } = request.params;
      const brief = await db.brief.findFirst({
        where: { shareToken },
      });
      if (!brief)
        return reply.code(404).send({ success: false, error: { code: 'NOT_FOUND', statusCode: 404 } });
      if (brief.shareTokenExpiresAt && brief.shareTokenExpiresAt < new Date())
        return reply.code(410).send({ success: false, error: { code: 'SHARE_LINK_EXPIRED', statusCode: 410 } });

      return reply.send({
        success: true,
        data: {
          sessionScore: brief.sessionScore,
          consistencyRate: brief.consistencyRate,
          topRecommendations: brief.topRecommendations,
          narrativeText: brief.narrativeText,
        },
      });
    }
  );
}

File: verdict-backend/app/routers/briefs.py
from datetime import datetime
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.ext.asyncio import AsyncSession
from sqlalchemy import select
from app.database import get_db
from app.middleware.auth import require_auth
from app.models.user import User
from app.models.brief import Brief

router = APIRouter()


@router.get("/share/{share_token}")
async def get_shared_brief(share_token: str, db: AsyncSession = Depends(get_db)):
    result = await db.execute(select(Brief).where(Brief.share_token == share_token))
    brief = result.scalar_one_or_none()
    if not brief:
        raise HTTPException(404, detail={"code": "NOT_FOUND"})
    if brief.share_token_expires_at and brief.share_token_expires_at < datetime.utcnow():
        raise HTTPException(410, detail={"code": "SHARE_LINK_EXPIRED"})
    return {
        "success": True,
        "data": {
            "sessionScore": brief.session_score,
            "consistencyRate": brief.consistency_rate,
            "topRecommendations": brief.top_recommendations,
            "narrativeText": brief.narrative_text,
        },
    }


@router.get("/{brief_id}")
async def get_brief(brief_id: str, db: AsyncSession = Depends(get_db), user: User = Depends(require_auth)):
    result = await db.execute(select(Brief).where(Brief.id == brief_id, Brief.firm_id == user.firm_id))
    brief = result.scalar_one_or_none()
    if not brief:
        raise HTTPException(404, detail={"code": "NOT_FOUND"})
    return {
        "success": True,
        "data": {
            "id": brief.id,
            "sessionScore": brief.session_score,
            "consistencyRate": brief.consistency_rate,
            "deltaVsBaseline": brief.delta_vs_baseline,
            "confirmedFlags": brief.confirmed_flags,
            "objectionCount": brief.objection_count,
            "composureAlerts": brief.composure_alerts,
            "topRecommendations": brief.top_recommendations,
            "narrativeText": brief.narrative_text,
            "weaknessMapScores": brief.weakness_map_scores,
            "shareToken": brief.share_token,
            "pdfS3Key": brief.pdf_s3_key,
            "createdAt": brief.created_at,
        },
    }
